backend/entityservice/tasks/encoding_uploading.py

```diff
@@ -56,12 +60,21 @@ def pull_external_data(project_id, dp_id,
     bucket_name = encoding_object_info['bucket']
     object_name = encoding_object_info['path']
 
-    stat, encodings_stream = stat_and_stream_object(bucket_name, object_name, parse_minio_credentials(encoding_credentials))
+    stat, encodings_stream = stat_and_stream_object(bucket_name, object_name, env_credentials)
     count = int(stat.metadata['X-Amz-Meta-Hash-Count'])
     size = int(stat.metadata['X-Amz-Meta-Hash-Size'])
     log.debug(f"Processing {count} encodings of size {size}")
     assert count == len(encoding_to_block_map), f"Expected {count} encodings in blocks got {len(encoding_to_block_map)}"
 
+    # load clks properly
+    encodings = json.loads(encodings_stream.data.decode())['clks']
+
+    stream = io.BytesIO()
+    for clk in encodings:
+        stream.write(deserialize_bytes(clk))
+    stream.seek(0)
+    encodings_stream = stream
+
```

Comment:
Previously, we grabbed one encoding at a time from the encodings_stream and wrote it straight to the database. Enriching it in between with a generator. This is very memory efficient, as it never has to hold the whole file in memory. 
With your approach, you first create a full json representation in memory, and then a second one as BytesIO. That is quite wasteful. Consider uploads of 100s of millions of entities. That's quite a few gigabytes of data!